src/ProxyManager/Exception/FileNotWritableException.php

```diff
@@ -30,4 +31,9 @@ public static function fromNotWritableDirectory(string $directory) : self
             $directory
         ));
     }
+
+    public static function exception(FileWriterException $e) : self
```

Comment:
I would probably call this `fromPrevious` or such.

If the other named constructors are no longer in use, they should be `@deprecated`.

Let's also rename `$e` to `$previous`